zzzz_modular_occulus/code/game/objects/items/devices/magnetic_lock.dm

```diff
@@ -0,0 +1,539 @@
+#define STATUS_INACTIVE 0
+#define STATUS_ACTIVE 1
+#define STATUS_BROKEN -1
+
+#define LAYER_ATTACHED 3.2
+#define LAYER_NORMAL 3
+
+/obj/item/device/magnetic_lock
+	name = "magnetic door lock"
+	desc = "A large, ID locked device used for completely locking down airlocks."
+	icon = 'zzzz_modular_occulus/icons/obj/magnetic_locks.dmi'
+	icon_state = "inactive_CENTCOM"
+	//icon_state = "inactive"
+	w_class = ITEM_SIZE_NORMAL
+	req_access = list(access_cent_specops)
+	health = 150
+
+	var/department = "CENTCOM"
+	var/status = 0
+	var/locked = 1
+	var/hacked = 0
+	var/invincible = FALSE
+	var/processpower = TRUE
+	var/constructionstate = 0
+	var/drain_per_second = 3
+	var/last_process_time = 0
+	var/obj/machinery/door/airlock/target_node1 = null
+	var/obj/machinery/door/airlock/target_node2 = null
+	var/obj/machinery/door/airlock/target = null
+	var/obj/item/weapon/cell/powercell
+	var/obj/item/weapon/cell/internal_cell
+	var/datum/effect/effect/system/spark_spread/spark_system
+
+/obj/item/device/magnetic_lock/security
+	department = "Security"
+	icon_state = "inactive_Security"
+	req_access = list(access_security)
+
+/obj/item/device/magnetic_lock/engineering
+	department = "Engineering"
+	icon_state = "inactive_Engineering"
+	req_access = null
+	req_one_access = list(access_engine_equip, access_atmospherics)
+/*
+/obj/item/device/magnetic_lock/security/legion
+	name = "legion magnetic door lock"
+	req_access = (access_legion)
+	w_class = ITEMSIZE_SMALL
+
+/obj/item/device/magnetic_lock/security/legion/Initialize()
+	. = ..()
+	desc = "A large, ID locked device used for completely locking down airlocks. This one carries the insignia of the Tau Ceti Foreign Legion."
+*/
+/obj/item/device/magnetic_lock/Initialize()
+	. = ..()
+
+	powercell = new /obj/item/weapon/cell/large/high()
+	internal_cell = new /obj/item/weapon/cell/large()
```

Comment:
The reasons are thus: 
The internal power cell prevents the lock from getting yanked off the wall the moment you yank out the cell, which would happen very fast otherwise.
It also provides a buffer for which you can remove the power cell with the lock still on an airlock, and add a new cell to replace it.

While it's not consistent with Eris code's overwhelming consistent code base, I think it provides enough of a benefit to have it in. It's also, not going to be apparent to be interacted with by the players in a manner that would require them to do anything with the internal cell itself such as removing, replacing, or recharging it. 

The hacking process ends when you disconnect the wires from the internal cell via wire cutters, but you can't actually physically touch the cell. It's forever locked away inside the frame, built in.
Functionally to the player it's as the brace really only uses one cell.